plugin.py

```diff
@@ -112,10 +124,33 @@ def run() -> None:
 
         sublime.set_timeout_async(run)
 
-    def m_dart_textDocument_publishClosingLabels(self, params: Any) -> None:
-        # TODO: Implement me.
-        pass
 
+
+    def apply_closing_labels(self, view, labels):
+        for label in labels:
+            final_character_position = label["range"]["end"]
+            if final_character_position["character"] < 1:
+                return
+            row = label["range"]["end"]["line"]
+            point = view.line(view.text_point(row, 0)).end()
+            view.add_phantom(closing_labels,
+                                sublime.Region(point, point+1),
+                                build_label(view, label["label"]),
+                                sublime.LAYOUT_INLINE)
+
+
+    def m_dart_textDocument_publishClosingLabels(self, params: Any) -> None:
+        session = self.weaksession()
+        if not session:
+            return
+        for sv in session.session_views_async():
+            if sv.view.is_valid():
+                if sv.view.file_name() not in params["uri"]:
+                    return
+                sv.view.erase_phantoms(closing_labels)
+                if session.config.init_options.get("closingLabels"):
+                    self.apply_closing_labels(sv.view, reversed(params["labels"]))
```

Comment:
There is already a method for getting the correct view/buffer:

```python
sb = session.get_session_buffer_for_uri_async(params["uri"])
if sb:
    for sv in sb.session_views:
        self.apply_closing_labels(sv, reversed(params["label"]))
```

Is checking for `session.config.init_options.get("closingLabels")` necessary? If it's in the initializationOptions, then apparently the user wanted to see closing labels right?

Reply:
Nice the code will be a bit smaller with that change!

I believe the check if necessary as `closingLabels` can be `false`, I recognize this part of the code feels a bit clumsy as I'm only checking *now* if the setting is set to true to draw the labels and before that I remove them.

Reply:
if we set `closingLabels` to `false` in initialization options, 
in that case I would expect that the lanuage server doesn't emit a `textDocument/publishClosingLabels` notifications. 

I also believe that the if seems unnecessary. :)